src/Library.java

```diff
@@ -0,0 +1,220 @@
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Library {
+    private List<Branch> branches;
+    private List<Patron> patrons;
+    private List<Reservation> reservations;
+    private Notification notification;
+    private RecommendationStrategy recommendationStrategy;
+
+    // Constructor
+    public Library(Notification notification, RecommendationStrategy recommendationStrategy) {
+        this.branches = new ArrayList<>();
+        this.patrons = new ArrayList<>();
+        this.reservations = new ArrayList<>();
+        this.notification = notification;
+        this.recommendationStrategy = recommendationStrategy;
+    }
+
+    // Branch management
+    public void addBranch(Branch branch) {
+        branches.add(branch);
+    }
+
+    public void transferBook(String isbn, String fromBranchId, String toBranchId) {
+        Branch fromBranch = findBranchById(fromBranchId);
+        Branch toBranch = findBranchById(toBranchId);
+        if (fromBranch != null && toBranch != null) {
+            Book book = findBookInBranch(isbn, fromBranch);
+            if (book != null) {
+                fromBranch.removeBook(book);
+                toBranch.addBook(book);
+            }
+        }
+    }
+
+    private Branch findBranchById(String branchId) {
+        for (Branch branch : branches) {
```

Comment:
This logic should come under the Branch class. Library should not be coupled on the internal logic for finding a branch from id.